src/main/java/core/basesyntax/model/Transaction.java

```diff
@@ -0,0 +1,70 @@
+package core.basesyntax.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Transaction {
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String operation;
+
+        Operation(String operation) {
+            this.operation = operation;
+        }
+
+        public String getOperation() {
+            return operation;
+        }
+
+        public void setOperation(String operation) {
+            this.operation = operation;
+        }
+
+        public static Operation getOperationFromString(String operation) {
+            Map<String, Operation> operationMap = new HashMap<>();
+            for (Operation enumOperation : Operation.values()) {
+                operationMap.put(enumOperation.getOperation(), enumOperation);
```

Comment:
don't need to fill the map here, it's better to fill it outside of the enum and this class or at least use a static block of initialization for such purpose

Reply:
Where is it better to initialize this map, in Main?